syntax = "proto3";

option go_package = "sigmaos/proc";
import "google/protobuf/timestamp.proto";
import "sigmap/sigmap.proto";

message ProcSeqno {
  uint64 epoch = 1;
  uint64 seqno = 2;
  string procqID = 3;
  string scheddID = 4;
}

message ProcEnvProto {
  string pidStr = 1;
  string program = 2;
  string realmStr = 3;
  Tprincipal principal = 4;
  string procDir = 5;
  string parentDir = 6;
  map<string, TendpointProto> etcdEndpoints = 7;
  string outerContainerIPStr = 8;
  string innerContainerIPStr = 9;
  string kernelID = 10;
  string buildTag = 11;
  string perf = 12;
  string debug = 13;
  string uprocdPIDStr = 14;
  bool privileged = 15;
  bool overlays = 16;
  int64 netFail = 17;
  int64 crash = 18;
  int64 partition = 19;
  int32 howInt = 20;
  google.protobuf.Timestamp spawnTimePB = 21;
  string strace = 22;
  TendpointProto scheddEndpointProto = 23;
  TendpointProto namedEndpointProto = 24;
  bool useSPProxy = 25;
  bool useNetProxy = 26;
  map<string, SecretProto> secretsMap = 27;
  repeated string sigmaPath = 28;
  repeated string kernels = 29;
  string realmSwitchStr = 30;
  string version = 31;
  string checkpointLocation = 32;
}

message ProcProto {
  ProcEnvProto procEnvProto = 1;
  repeated string args = 2;
  map<string, string> env = 3;
  uint32 typeInt = 4;
  uint32 mcpuInt = 5;
  uint32 memInt = 6;
  string sharedTarget = 7;
}
